### ISSUE TITLE
Cluster name invalid - Create failed

### ISSUE THREAD

**Issue by mjudeikis** [Azure/ARO-RP#671]
```
ERRO[2020-05-13T08:42:55+01:00]pkg/backend/openshiftcluster.go:196 backend.(*openShiftClusterBackend).updateAsyncOperation.func1() metadata.name: Invalid value: "40wwsy2z": cluster name must begin with a lower-case letter  client_principal_name= client_request_id=4f65a528-94ed-11ea-93d5-b46921811e1f component=backend correlation_id= request_id=066da2cb-7d8b-4cfa-8766-ef9e4d4da0c6 resource_group=v4-eastus resource_id=/subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourcegroups/v4-eastus/providers/microsoft.redhatopenshift/openshiftclusters/mjudeikis2 resource_name=mjudeikis2 subscription_id=225e02bc-43d0-43d1-a01a-17e584a4ef69


```

**Comment by jim-minter** [comment]
need to restrict RxDomainName + add tests + restrict random_id in custom.py

**Comment by m1kola** [comment]
@troy0820 can we close this one? I think it was fixed in #676?

**Comment by troy0820** [comment]
Yeah, I am okay with closing it.  This did get fixed in #676 and the tests have been passing.  @jim-minter okay to close?
